### PR TITLE
Reworks Flashbangs

### PR DESCRIPTION
**Pull request by ItsMarmite**
## What Does This PR Do
**Reworks Flashbangs**

Flashbangs have been reworked as follows:

- The weaken condition has been removed, flashbangs no longer weaken those within their area of effect.

- Applies a duration, up to a maximum of 10 seconds of KnockDown (crawling) to those in radius scaling depending on distance from the flashbang.

- Adds a duration, up to a maximum of 20 seconds, of Confusion (as if flashed) to those in a radius scaling similarly depending on distance from the flashbang.

## Why It's Good For The Game
Requested change. Moving away from hard stuns and towards stamina/crawling based combat. No more AOE up to 20 second stunbaton.

## Images of changes
[https://i.imgur.com/IbajtKz.mp4](url)

## Changelog
:cl:
tweak: Flashbangs now confuse and knockdown instead of stun
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/grenades/flashbang.dm [hunk 1]
```diff
@@ -47,20 +47,20 @@
...
 
 		var/distance = max(1, get_dist(source_turf, get_turf(M)))
-		var/stun_amount = max(20 SECONDS / distance, 6 SECONDS)
+		var/status_duration = max(10 SECONDS / distance, 4 SECONDS)
 
 		// Flash
 		if(flash)
 			if(M.flash_eyes(affect_silicon = TRUE))
-				M.Weaken(stun_amount)
+				M.Confused(status_duration)*2
```
Comment (hal9000PR) [inline code/game/objects/items/weapons/grenades/flashbang.dm:55]:
```suggestion
				M.Confused(status_duration * 2)
```

**Comment by KittyNoodle** [comment]
the cl is really bare maybe change it to this?

> tweak: Flashbangs now confuse and knockdown instead of stun

**Comment by lewcc** [comment]
Might it still maybe make sense to have the flashbang inflict weaken on you (maybe not for the full 10 seconds) if it goes off in your hand? It's still a somewhat concussive grenade, and might make sense to knock em out if someone's trying to game hard.

**Comment by ItsMarmite** [comment]
> the cl is really bare maybe change it to this?
> 
> > tweak: Flashbangs now confuse and knockdown instead of stun



> the cl is really bare maybe change it to this?
> 
> > tweak: Flashbangs now confuse and knockdown instead of stun

Edited to fit the suggestion.

**Comment by ItsMarmite** [comment]
> Might it still maybe make sense to have the flashbang inflict weaken on you (maybe not for the full 10 seconds) if it goes off in your hand? It's still a somewhat concussive grenade, and might make sense to knock em out if someone's trying to game hard (plus: it's at least a little funny).



> Might it still maybe make sense to have the flashbang inflict weaken on you (maybe not for the full 10 seconds) if it goes off in your hand? It's still a somewhat concussive grenade, and might make sense to knock em out if someone's trying to game hard (plus: it's at least a little funny).

The intention is to move away from stun based combat. Being stunned for up to 20 seconds, as it currently is, is not a little funny.

**Comment by Inadvizable** [comment]
> Might it still maybe make sense to have the flashbang inflict weaken on you (maybe not for the full 10 seconds) if it goes off in your hand? It's still a somewhat concussive grenade, and might make sense to knock em out if someone's trying to game hard (plus: it's at least a little funny).

There is however a problem in that: gamers will abuse everything, and it won't take long for them to figure out they can drop a bang on top of someone to prevent them from getting up. I'm not sure how big of a problem that would be, but I am entertaining this possibility. They already do dumb stuff like drag a primed flashbang...

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**